cty/function/stdlib/collection.go

```diff
@@ -299,20 +299,52 @@ var ContainsFunc = function.New(&function.Spec{
 		},
 	},
 	Type: function.StaticReturnType(cty.Bool),
-	Impl: func(args []cty.Value, retType cty.Type) (ret cty.Value, err error) {
+	Impl: func(args []cty.Value, retType cty.Type) (cty.Value, error) {
 		arg := args[0]
 		ty := arg.Type()
 
 		if !ty.IsListType() && !ty.IsTupleType() && !ty.IsSetType() {
 			return cty.NilVal, errors.New("argument must be list, tuple, or set")
 		}
 
-		_, err = Index(cty.TupleVal(arg.AsValueSlice()), args[1])
-		if err != nil {
-			return cty.False, nil
+		if args[0].IsNull() {
+			return cty.NilVal, errors.New("cannot search a nil list or set")
+		}
+
+		if args[0].LengthInt() == 0 {
+			return cty.NilVal, errors.New("cannot search an empty list or set")
```

Comment:
Yeah, this was specifically a carry-over from terraform. I think we can even accept this directly in terraform, since there's no working case that would be relying on the error.

Reply:
It looks like this check was only ever in Terraform _because_ it was using the `Index` function internally, so by writing out the implementation directly here we can "fix the glitch"! :upside_down_face: